Chirp/Pages/RegistrationPage/RegistrationViewModel.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  RegistrationViewModel.swift
+//  Chirp
+//
+//  Created by Stepan Kukharskyi on 6/9/23.
+//
+
+import UIKit
+import Combine
+import FirebaseAuth
+
+class RegistrationViewModel {
+    
+    private var phoneNumber: String
+    private var imageExtension: String = "jpg"
+    
+    private let service = NetworkService.shared
+    private var cancellables: Set<AnyCancellable> = []
+    var didRegisterUser = PassthroughSubject<Bool,Never>()
+    
+    init(phoneNumber: String) {
+        self.phoneNumber = phoneNumber
+    }
+    
+    
+    func createAccount(name: String, profileImage: UIImage?) {
+        guard let userID = Auth.auth().currentUser?.uid else {
+            AlertToast.showAlert(message: AppError.registrationFailed.errorDescription ?? "Error", type: .error)
+            return
+        }
+        
+        LoaderView.shared.show(message: "Creating Account...")
+        // try saving image first
+        if let image = profileImage, let imageData = image.jpegData(compressionQuality: 0.8) {
+            let request = UploadProfileImageRequest(userID: userID, imageData: imageData,
+                                                               imageExtension: imageExtension,
+                                                               timestamp: Date().getCurrentTimestamp())
+            service.uploadProfileImage(request: request)
+                .sink { response in
+                switch response {
+                case .failure(let error):
+                    AlertToast.showAlert(message: error.localizedDescription, type: .error)
```

Comment:
You also want to hide the loader if there's an error because it does not proceed after this error